cmake/CPM.cmake

```diff
@@ -271,6 +271,55 @@ function(cpm_check_if_package_already_added CPM_ARGS_NAME CPM_ARGS_VERSION)
           "${CPM_INDENT} requires a newer version of ${CPM_ARGS_NAME} (${CPM_ARGS_VERSION}) than currently included (${CPM_PACKAGE_VERSION})."
       )
     endif()
+
+    if(CPM_ARGS_OPTIONS)
```

Comment:
What if when the that package was first added options were not set and not they are, or vice-versa?

Reply:
Yes maybe this cases are not well checked.